### PR TITLE
fix the arguments for Values

### PR DESCRIPTION
**Pull request by zzrcxb**
Please refer to issue #12

### CODE REVIEW

File: src/bap/bir.py [hunk 1]
```diff
@@ -253,7 +253,8 @@ class Values(Map) :
     It is a mapping from the tid of a preceeding block,
     to an expression that denotes a value.
     """
-    pass
+    def __init__(self, *args):
+        super().__init__([ADT(p) for p in args[0]])
```
Comment (ivg) [inline src/bap/bir.py]: First of all, `super` without args works only in Python 3.x and we are still striving to support both versions. 

Next, I think a more efficient and more robust solution would be to overload the `Map` constructor and just initialize `self.elements`, e.g., something like this:
```python
class Values(Map):
    def __init__(self, *args):
        super(Map,self).__init__(args)
        self.elements = dict(args[0])
```

What do you think?

**Review by ivg — CHANGES_REQUESTED** [review]

File: src/bap/bir.py (hunk 1 above)
Comment (zzrcxb) [reply src/bap/bir.py]: Thanks for your reply! If you want to support both Python versions, I agree.
I was trying to trick Map.\_\_init\_\_. But your way seems neater.

File: src/bap/bir.py [hunk 2]
```diff
@@ -253,7 +253,9 @@ class Values(Map) :
     It is a mapping from the tid of a preceeding block,
     to an expression that denotes a value.
     """
-    pass
+    def __init__(self, *args):
+        super(Map,self).__init__(args)
```
Comment (ivg) [inline src/bap/bir.py]: pylint doesn't like it, for some reason it believes that the first parameter of `super` should _always_ be the current class. I think we can try to hush it with `# pylint: disable=bad-super-call` added to this line. Alternatively, we can just call the `ADT` constructor directly, e.g., `ADT.__init__(args)`

**Review by ivg — CHANGES_REQUESTED** [review]

File: src/bap/bir.py (hunk 2 above)
Comment (zzrcxb) [reply src/bap/bir.py]: sometimes I just ignore pylint's warning like this, lol. Maybe it's not a very good habit.

**Comment by ivg** [comment]
Thanks for your work!

**Pull request merged**